bftengine/src/preprocessor/PreProcessor.hpp

```diff
@@ -175,6 +176,42 @@ class PreProcessor {
   concordUtil::Timers::Handle requestsStatusCheckTimer_;
   const uint64_t preExecReqStatusCheckPeriodMilli_;
   concordUtil::Timers &timers_;
+
+  // 5 Minutes , 300 seconds
+  static constexpr int64_t MAX_VALUE_MICROSECONDS = 300000000;
+
+  using Recorder = concord::diagnostics::Recorder;
+  struct Recorders {
+    Recorders() {
+      auto &registrar = concord::diagnostics::RegistrarSingleton::getInstance();
+      try {
+        registrar.perf.registerComponent("pre-execution",
+                                         {{"onMessage", onMessage},
+                                          {"launchReqPreProcessing", launchReqPreProcessing},
+                                          {"handlePreProcessedReqByNonPrimary", handlePreProcessedReqByNonPrimary},
+                                          {"handlePreProcessedReqByPrimary", handlePreProcessedReqByPrimary},
+                                          {"sendPreProcessRequestToAllReplicas", sendPreProcessRequestToAllReplicas},
+                                          {"finalizePreProcessing", finalizePreProcessing}});
+      } catch (std::invalid_argument &e) {
+        // if component already exists lets keep record on the same histograms
```

Comment:
I didn't notice this in my initial review. This doesn't work as intended. Instead, measurements will be silently dropped as I noted when updating the API in [this commit](https://github.com/vmware/concord-bft/pull/949/commits/b38c5fd850df55b70db95b46478947593488eb81) in [the tls pr](https://github.com/vmware/concord-bft/pull/949).

Unfortunately this is a bug that causes tests to fail from dual registration. I'm attempting to fix it, but the change might require input from you @Liorgal28 as it's a semantic issue. 

In short, there can only be one component with a given name. Ignoring the exception is no longer going to be an option because the process will terminate.

Reply:
I actually was able to fix this by modifying the test to clear the registrations. This is in general a problem of singletons, in that they introduce these types of side effects. Maybe in the future when we have a better startup sequence we can pass the registrar to components rather than using a singleton. For now this works fine and is easy enough.

Reply:
Thanks for addressing that issue, I commented in https://github.com/vmware/concord-bft/pull/949.